internal/internal_task_pollers_test.go

```diff
@@ -54,3 +68,132 @@ func TestLocalActivityPanic(t *testing.T) {
 	assert.Contains(t, perr.StackTrace(), "panic")
 	assert.Contains(t, perr.StackTrace(), t.Name(), "should mention the source location of the local activity that panicked")
 }
+
+func TestRespondTaskCompleted_failed(t *testing.T) {
+	t.Run("fail sends RespondDecisionTaskFailedRequest", func(t *testing.T) {
+		testTaskToken := []byte("test-task-token")
+
+		poller, client, _, _ := buildWorkflowTaskPoller(t)
+		client.EXPECT().RespondDecisionTaskFailed(gomock.Any(), &s.RespondDecisionTaskFailedRequest{
+			TaskToken:      testTaskToken,
+			Cause:          s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure.Ptr(),
+			Details:        []byte(assert.AnError.Error()),
+			Identity:       common.StringPtr(_testIdentity),
+			BinaryChecksum: common.StringPtr(getBinaryChecksum()),
+		}, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+
+		res, err := poller.RespondTaskCompleted(nil, assert.AnError, &s.PollForDecisionTaskResponse{
+			TaskToken: testTaskToken,
+			Attempt:   common.Int64Ptr(0),
+		}, time.Now())
+		assert.NoError(t, err)
+		assert.Nil(t, res)
+	})
+	t.Run("fail fails to send RespondDecisionTaskFailedRequest", func(t *testing.T) {
+		testTaskToken := []byte("test-task-token")
+
+		poller, client, _, _ := buildWorkflowTaskPoller(t)
+		client.EXPECT().RespondDecisionTaskFailed(gomock.Any(), &s.RespondDecisionTaskFailedRequest{
+			TaskToken:      testTaskToken,
+			Cause:          s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure.Ptr(),
+			Details:        []byte(assert.AnError.Error()),
+			Identity:       common.StringPtr(_testIdentity),
+			BinaryChecksum: common.StringPtr(getBinaryChecksum()),
+		}, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(assert.AnError)
+
+		// We cannot test RespondTaskCompleted since it uses backoff and has a hardcoded retry mechanism for 60 seconds.
+		_, err := poller.respondTaskCompletedAttempt(errorToFailDecisionTask(testTaskToken, assert.AnError, _testIdentity), &s.PollForDecisionTaskResponse{
+			TaskToken: testTaskToken,
+			Attempt:   common.Int64Ptr(0),
+		})
+		assert.ErrorIs(t, err, assert.AnError)
+	})
+	t.Run("fail skips sending for not the first attempt", func(t *testing.T) {
+		poller, _, _, _ := buildWorkflowTaskPoller(t)
+
+		res, err := poller.RespondTaskCompleted(nil, assert.AnError, &s.PollForDecisionTaskResponse{
+			Attempt: common.Int64Ptr(1),
+		}, time.Now())
+		assert.NoError(t, err)
+		assert.Nil(t, res)
+	})
+}
+
+func TestRespondTaskCompleted_Unsupported(t *testing.T) {
+	poller, _, _, _ := buildWorkflowTaskPoller(t)
+
+	assert.Panics(t, func() {
+		_, _ = poller.RespondTaskCompleted(assert.AnError, nil, &s.PollForDecisionTaskResponse{}, time.Now())
```

Comment:
given how many things this could be, probably worth adding a
```
PanicsWithValue(t, "unknown request type from ProcessWorkflowTask()", func() {...})
```